### DISCUSSION TITLE
Include file from path

### DISCUSSION THREAD

**Discussion by keriksson-rosenqvist-oqc** [python-poetry/poetry#8592] [Q&A] [answered]
When defining the packages in a pyproject.toml file, you can specify a `from` path for where to fin the source files, this doesn't seem to work the same with the `include` and `exclude` lists though. Is there a way to achieve this?

I want to achieve a built package that includes only some of the files from a patch but located at the top level.

Example pyproject.toml file
```
[tool.poetry]
name = "my_package"
version = "1.0.0"
description = "Some package including files from paths"
authors = ["Me"]
packages = [{include = "my_package", from = "src"}]
license = "BSD-3-Clause"
exclude = [{path = "tests", from = "src/my_package"}]
include = [{path = "tests/user_tests", from = "src/my_package"}]

[tool.poetry.dependencies]
python = ">=3.8.1,<3.11"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

That would generate this package
```
my_package-1.0.0
├──  my_package
│  ├── ....
│  └── __init__.py
└──  tests
   └── user_tests
       ├── ....
       └── __init__.py

```

Excluding all other directories and files in the `tests` directory, and placing the included `tests` subset at the root level.

**Comment by dimbleby** [comment]
read the docs https://python-poetry.org/docs/pyproject#include-and-exclude

**Reply by keriksson-rosenqvist-oqc** [reply]
I take it you're saying, It is not possible/supported.

**Reply by dimbleby** [reply]
I am saying that the documentation tells you what is possible/supported.

**Comment by skynavga** [comment]
Is there someone who actually knows an answer to the above question? Namely, how to specify an inclusion is relative to a `from` qualifier? For example,

```toml
[tool.poetry]
...
include = [ { path = "resources", from = "src" } ]
```

**Reply by keriksson-rosenqvist-oqc** [reply]
> This worked fine for me, after trying them all:
> 
> ```
> [tool.poetry]
> ...
> packages = [
>     { include = "my_lib", from = "libs/common" },
> ]
> ```

To clarify, the question is about doing this to non-`packages` includes/excludes.

**Reply by skynavga** [reply]
Exactly. Therefore, the original question has not been answered, namely, how to specify **non-package** relative inclusions/exclusions.

**Reply by radoering** [reply, marked as answer]
The answer is: "You can't" (until someone implements `from` and `to` for `include` / `exclude`, which should not be too difficult because there already is an implementation for `packages`)

**Reply by skynavga** [reply]
Thanks @radoering! That's what I expected, but just wanted the _true word_.

**Reply by keriksson-rosenqvist-oqc** [reply]
@skynavga Thank you for the post that lead to clear answer! :slightly_smiling_face:. Though the comment I quoted has apparently been deleted after I read it. :sweat_smile: 
@radoering Thanks for the clear confirmation. :+1:
